### PR TITLE
Add string format recommendation for Python

### PR DESCRIPTION
**Pull request by imlazyone**
Use `format()` instead of `%`

### CODE REVIEW

File: language-python.md [hunk 1]
```diff
@@ -400,3 +400,19 @@ If a function argument's name clashes with a reserved keyword, it is generally b
...
+#### String format
+
+We recommend to use `format()` instead of `%`.
+
+Recommend:
+```python
+'{} {}'.format('one', 'two')
+```
+
+Not recommend:
```
Comment (agirivera) [inline language-python.md]: ? _Recommended_ and _not recommended_?

File: language-python.md [hunk 2]
```diff
@@ -400,3 +400,19 @@ If a function argument's name clashes with a reserved keyword, it is generally b
...
+```python
+this `'%s %s' % ('one', 'two')`
```
Comment (agirivera) [inline language-python.md]: ? What is _this_ for?

Comment (arkhi) [inline language-python.md]: https://twitter.com/bendhalpern/status/578925947245633536

File: language-python.md [hunk 3]
```diff
@@ -400,3 +400,19 @@ If a function argument's name clashes with a reserved keyword, it is generally b
...
 
 Constants are usually defined on a module level and written in all capital letters with underscores separating words. Examples include `MAX_OVERFLOW` and `TOTAL` .
+
+#### String format
+
+We recommend to use `format()` instead of `%`.
+
+Recommend:
+```python
+'{} {}'.format('one', 'two')
```
Comment (arkhi) [inline language-python.md]: I’m not familiar with Python, Should there be something in `{}` or are these just generic placeholders?

I would expect `{one}` and `{two}`. :)

Comment (imlazyone) [inline language-python.md]: Yes I think you can but it's not necessary.

**Comment by imlazyone** [comment]
Updated. Please review again.

**Comment by imlazyone** [comment]
@agirivera and @arkhi are you good with the changes?

**Comment by imlazyone** [comment]
Sure, `format()` is the new way and `%` is the old way, in the new way it supports more flexible ways to manipulate the string, but however lots of programmers still prefer the old way because it's less typing work :)

Here's an article: https://pyformat.info/

**Comment by arkhi** [comment]
Sorry, I meant to have it in the file you updated. :)

If you want to update it, you can link directly to https://pyformat.info/#simple. If you don’t, feel free to merge.

Thanks for the explanation in any case!

**Comment by HouCoder** [comment]
The `format` function makes the code easier to read and the code more like a programming language even I'm not a Python dev.

**Comment by agirivera** [comment]
:+1: Looks good to go

**Pull request merged**